File: packages/java/typescript-generator/src/test/java/com/vaadin/hilla/parser/plugins/nonnull/superclassmethods/PersonEndpoint.java
package com.vaadin.hilla.parser.plugins.nonnull.superclassmethods;

import com.vaadin.hilla.parser.testutils.annotations.Endpoint;

@Endpoint
public class PersonEndpoint extends CrudEndpoint<PersonEndpoint.Person, Integer>
        implements PagedData<PersonEndpoint.Person> {

    public static class Person {
        private String name;

        public String getName() {
            return name;
        }

        public void setName(String name) {
            this.name = name;
        }
    }
}

File: packages/java/typescript-generator/src/test/kotlin/com/vaadin/hilla/parser/plugins/nonnull/kotlin/superclasses/javaendpointexposed/PersonEndpoint.kt
package com.vaadin.hilla.parser.plugins.nonnull.kotlin.superclasses.javaendpointexposed

import com.vaadin.hilla.parser.testutils.annotations.Endpoint

@Endpoint
class PersonEndpoint : CrudEndpoint<PersonEndpoint.Person, Int>(), PagedData<PersonEndpoint.Person> {

    class Person {
        var name: String? = null
    }

    fun create(entity: Person): Int {
        return 0
    }
}
